spec/controllers/teams_controller_spec.rb

```diff
@@ -146,30 +146,52 @@
       describe "with valid params" do
         it "updates the requested team" do
           expect_any_instance_of(Team).to receive(:update_attributes).with(ActionController::Parameters.new({ 'name' => 'Blue' }).permit(:name))
-          put :update, params: { id: team.to_param, team: { 'name' => 'Blue' } }
+          patch :update, params: { id: team.to_param, team: { 'name' => 'Blue' } }
         end
 
         it "assigns the requested team as @team" do
-          put :update, params: { id: team.to_param, team: valid_attributes }
+          patch :update, params: { id: team.to_param, team: valid_attributes }
           expect(assigns(:team)).to eq(team)
         end
 
         it "redirects to the team" do
-          put :update, params: { id: team.to_param, team: valid_attributes }
+          patch :update, params: { id: team.to_param, team: valid_attributes }
           expect(response).to redirect_to(team)
         end
+
+        context 'with nested role attributes' do
+          let(:github_handle) { valid_attributes[:roles_attributes].first[:github_handle] }
+
+          it 'creates a new user from github_handle' do
+            expect {
+              patch :update, params: { id: team.to_param, team: valid_attributes }
+            }.to change { User.count }.by 1
+
+            expect(User.last.github_handle).to eql github_handle
+          end
+
+          it 'finds an existing user by case-insenstive github_handle' do
+            existing_user = create :user, github_handle: github_handle.upcase
+            expect {
```

Comment:
I just wanted to have a different case. I thought of adding a lambda `randomize_case` but decided against it. Do you think it would make the intention of the example more clear? If so, I'll add that.